From 2d1c3ed6046842f1b30dd76c3f0f9811973629c5 Mon Sep 17 00:00:00 2001
From: Manos Koukoutos <manoskouk@chromium.org>
Date: Mon, 18 Jul 2022 10:54:58 +0000
Subject: [PATCH] [test] Remove some unused includes

Mostly test/cctest/.

Bug: v8:13006
Change-Id: I8853d38feb79bed6234a4354ab25a13255a1871b
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3755149
Commit-Queue: Manos Koukoutos <manoskouk@chromium.org>
Reviewed-by: Jakob Kummerow <jkummerow@chromium.org>
Cr-Commit-Position: refs/heads/main@{#81777}
---
 test/cctest/assembler-helper-arm.cc               |  2 --
 test/cctest/cctest.cc                             |  1 -
 test/cctest/cctest.h                              |  4 +---
 test/cctest/test-api-icu.cc                       |  1 -
 test/cctest/test-api-interceptors.cc              |  3 ---
 test/cctest/test-api.cc                           |  8 --------
 test/cctest/test-api.h                            |  2 --
 test/cctest/test-assembler-arm.cc                 |  2 --
 test/cctest/test-assembler-arm64.cc               |  5 +----
 test/cctest/test-assembler-ia32.cc                |  4 +---
 test/cctest/test-assembler-x64.cc                 |  1 -
 test/cctest/test-code-stub-assembler.cc           |  7 ++-----
 test/cctest/test-cpu-profiler.cc                  |  8 ++------
 test/cctest/test-debug-helper.cc                  |  1 -
 test/cctest/test-debug.cc                         |  3 ---
 test/cctest/test-disasm-mips.cc                   |  5 -----
 test/cctest/test-feedback-vector.cc               |  8 ++------
 test/cctest/test-feedback-vector.h                |  1 +
 test/cctest/test-field-type-tracking.cc           |  5 +----
 test/cctest/test-func-name-inference.cc           |  2 --
 test/cctest/test-heap-profiler.cc                 |  1 -
 test/cctest/test-helper-riscv64.cc                |  1 -
 test/cctest/test-helper-riscv64.h                 |  3 ---
 test/cctest/test-ignition-statistics-extension.cc |  3 ++-
 test/cctest/test-inobject-slack-tracking.cc       |  3 +--
 test/cctest/test-inspector.cc                     |  1 -
 test/cctest/test-javascript-arm64.cc              | 10 ----------
 test/cctest/test-js-arm64-variables.cc            | 10 ----------
 test/cctest/test-js-to-wasm.cc                    |  2 --
 test/cctest/test-js-weak-refs.cc                  |  1 -
 test/cctest/test-liveedit.cc                      |  1 -
 test/cctest/test-local-handles.cc                 |  4 ----
 test/cctest/test-lockers.cc                       |  5 -----
 test/cctest/test-log-stack-tracer.cc              |  3 ---
 test/cctest/test-macro-assembler-arm.cc           |  1 -
 test/cctest/test-macro-assembler-arm64.cc         |  4 ----
 test/cctest/test-macro-assembler-loong64.cc       |  1 -
 test/cctest/test-macro-assembler-mips.cc          |  2 --
 test/cctest/test-macro-assembler-mips64.cc        |  1 -
 test/cctest/test-macro-assembler-riscv64.cc       |  3 ---
 test/cctest/test-macro-assembler-x64.cc           |  2 --
 test/cctest/test-orderedhashtable.cc              |  1 -
 test/cctest/test-parsing.cc                       |  5 -----
 test/cctest/test-profile-generator.cc             |  1 -
 test/cctest/test-random-number-generator.cc       |  4 +---
 test/cctest/test-serialize.cc                     |  7 -------
 test/cctest/test-shared-strings.cc                |  1 -
 test/cctest/test-smi-lexicographic-compare.cc     |  1 -
 test/cctest/test-stack-unwinding-win64.cc         |  3 +--
 test/cctest/test-strings.cc                       |  3 ---
 test/cctest/test-swiss-name-dictionary-infra.h    |  4 ----
 test/cctest/test-sync-primitives-arm.cc           |  8 ++------
 test/cctest/test-sync-primitives-arm64.cc         |  5 +----
 test/cctest/test-temporal-parser.cc               |  2 --
 test/cctest/test-transitions.cc                   |  7 +++----
 test/cctest/test-typedarrays.cc                   |  7 +------
 test/cctest/test-unscopables-hidden-prototype.cc  |  1 -
 test/cctest/test-usecounters.cc                   |  2 --
 test/cctest/test-utils-arm64.cc                   |  3 ---
 test/cctest/test-utils-arm64.h                    |  3 ---
 test/cctest/test-utils.cc                         |  3 ---
 test/cctest/trace-extension.h                     |  1 -
 test/common/wasm/wasm-macro-gen.h                 |  2 --
 63 files changed, 23 insertions(+), 186 deletions(-)

diff --git a/test/cctest/assembler-helper-arm.cc b/test/cctest/assembler-helper-arm.cc
index e4a79f2f888..ca4041dc723 100644
--- a/test/cctest/assembler-helper-arm.cc
+++ b/test/cctest/assembler-helper-arm.cc
@@ -6,8 +6,6 @@
 
 #include "src/codegen/macro-assembler.h"
 #include "src/execution/isolate-inl.h"
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/cctest.cc b/test/cctest/cctest.cc
index da1a91cca05..a04d80111b8 100644
--- a/test/cctest/cctest.cc
+++ b/test/cctest/cctest.cc
@@ -45,7 +45,6 @@
 #include "src/codegen/optimized-compilation-info.h"
 #include "src/common/globals.h"
 #include "src/compiler/pipeline.h"
-#include "src/debug/debug.h"
 #include "src/flags/flags.h"
 #include "src/objects/objects-inl.h"
 #include "src/trap-handler/trap-handler.h"
diff --git a/test/cctest/cctest.h b/test/cctest/cctest.h
index b8f73a059a1..031626ce29c 100644
--- a/test/cctest/cctest.h
+++ b/test/cctest/cctest.h
@@ -35,11 +35,9 @@
 #include "src/base/enum-set.h"
 #include "src/codegen/register-configuration.h"
 #include "src/debug/debug-interface.h"
-#include "src/execution/isolate.h"
+#include "src/execution/isolate-inl.h"
 #include "src/execution/simulator.h"
-#include "src/flags/flags.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/objects/js-function.h"
 #include "src/objects/objects.h"
 #include "src/zone/accounting-allocator.h"
diff --git a/test/cctest/test-api-icu.cc b/test/cctest/test-api-icu.cc
index e251d2c0c70..23b683399a6 100644
--- a/test/cctest/test-api-icu.cc
+++ b/test/cctest/test-api-icu.cc
@@ -8,7 +8,6 @@
 
 #include "include/v8-isolate.h"
 #include "include/v8-local-handle.h"
-#include "src/objects/objects-inl.h"
 #include "test/cctest/cctest.h"
 #include "unicode/locid.h"
 
diff --git a/test/cctest/test-api-interceptors.cc b/test/cctest/test-api-interceptors.cc
index 9be7769d6a1..2a7f2528dc1 100644
--- a/test/cctest/test-api-interceptors.cc
+++ b/test/cctest/test-api-interceptors.cc
@@ -8,15 +8,12 @@
 
 #include "include/v8-function.h"
 #include "src/api/api-inl.h"
-#include "src/base/platform/platform.h"
 #include "src/codegen/compilation-cache.h"
-#include "src/execution/arguments.h"
 #include "src/execution/execution.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/objects.h"
 #include "src/runtime/runtime.h"
 #include "src/strings/unicode-inl.h"
-#include "src/utils/utils.h"
 #include "test/cctest/test-api.h"
 
 using ::v8::Context;
diff --git a/test/cctest/test-api.cc b/test/cctest/test-api.cc
index 154c5af2274..e859dfb9fd9 100644
--- a/test/cctest/test-api.cc
+++ b/test/cctest/test-api.cc
@@ -56,32 +56,24 @@
 #include "src/base/strings.h"
 #include "src/codegen/compilation-cache.h"
 #include "src/compiler/globals.h"
-#include "src/debug/debug.h"
-#include "src/execution/arguments.h"
 #include "src/execution/execution.h"
 #include "src/execution/futex-emulation.h"
 #include "src/execution/protectors-inl.h"
-#include "src/execution/vm-state.h"
 #include "src/handles/global-handles.h"
-#include "src/heap/evacuation-allocator.h"
 #include "src/heap/heap-inl.h"
 #include "src/heap/incremental-marking.h"
 #include "src/logging/metrics.h"
 #include "src/objects/feedback-vector-inl.h"
 #include "src/objects/feedback-vector.h"
 #include "src/objects/hash-table-inl.h"
-#include "src/objects/js-array-buffer-inl.h"
 #include "src/objects/js-array-inl.h"
 #include "src/objects/js-promise-inl.h"
 #include "src/objects/lookup.h"
 #include "src/objects/map-updater.h"
-#include "src/objects/module-inl.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/string-inl.h"
 #include "src/objects/synthetic-module-inl.h"
 #include "src/profiler/cpu-profiler.h"
-#include "src/profiler/symbolizer.h"
-#include "src/strings/unicode-inl.h"
 #include "src/utils/utils.h"
 #include "test/cctest/heap/heap-tester.h"
 #include "test/cctest/heap/heap-utils.h"
diff --git a/test/cctest/test-api.h b/test/cctest/test-api.h
index 56128236dba..48f3cd2c44a 100644
--- a/test/cctest/test-api.h
+++ b/test/cctest/test-api.h
@@ -5,8 +5,6 @@
 #ifndef V8_TEST_CCTEST_TEST_API_H_
 #define V8_TEST_CCTEST_TEST_API_H_
 
-#include "src/init/v8.h"
-
 #include "src/api/api.h"
 #include "src/execution/isolate.h"
 #include "src/execution/vm-state.h"
diff --git a/test/cctest/test-assembler-arm.cc b/test/cctest/test-assembler-arm.cc
index 9dc2bad84fb..65dee3a07a3 100644
--- a/test/cctest/test-assembler-arm.cc
+++ b/test/cctest/test-assembler-arm.cc
@@ -31,10 +31,8 @@
 #include "src/base/utils/random-number-generator.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
-#include "src/diagnostics/disassembler.h"
 #include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/assembler-helper-arm.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-assembler-arm64.cc b/test/cctest/test-assembler-arm64.cc
index 71698569606..ef88b2963c7 100644
--- a/test/cctest/test-assembler-arm64.cc
+++ b/test/cctest/test-assembler-arm64.cc
@@ -28,12 +28,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+
 #include <cmath>
 #include <limits>
 
-#include "src/init/v8.h"
-
-#include "src/base/platform/platform.h"
 #include "src/base/utils/random-number-generator.h"
 #include "src/codegen/arm64/assembler-arm64-inl.h"
 #include "src/codegen/arm64/decoder-arm64-inl.h"
@@ -42,7 +40,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/diagnostics/arm64/disasm-arm64.h"
 #include "src/execution/arm64/simulator-arm64.h"
-#include "src/execution/simulator.h"
 #include "src/heap/factory.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/test-utils-arm64.h"
diff --git a/test/cctest/test-assembler-ia32.cc b/test/cctest/test-assembler-ia32.cc
index cea8e682e8f..962fd142b4f 100644
--- a/test/cctest/test-assembler-ia32.cc
+++ b/test/cctest/test-assembler-ia32.cc
@@ -33,12 +33,10 @@
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
-#include "src/diagnostics/disassembler.h"
+#include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
-#include "test/common/assembler-tester.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-assembler-x64.cc b/test/cctest/test-assembler-x64.cc
index 231fee4b138..dafdd40866b 100644
--- a/test/cctest/test-assembler-x64.cc
+++ b/test/cctest/test-assembler-x64.cc
@@ -36,7 +36,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-code-stub-assembler.cc b/test/cctest/test-code-stub-assembler.cc
index 6694fb64315..2dd3ebcf2ae 100644
--- a/test/cctest/test-code-stub-assembler.cc
+++ b/test/cctest/test-code-stub-assembler.cc
@@ -10,7 +10,7 @@
 #include "src/builtins/builtins-promise-gen.h"
 #include "src/builtins/builtins-promise.h"
 #include "src/builtins/builtins-string-gen.h"
-#include "src/codegen/code-factory.h"
+#include "src/builtins/builtins-utils-inl.h"
 #include "src/codegen/code-stub-assembler.h"
 #include "src/codegen/interface-descriptors-inl.h"
 #include "src/compiler/node.h"
@@ -18,15 +18,12 @@
 #include "src/execution/isolate.h"
 #include "src/heap/heap-inl.h"
 #include "src/numbers/hash-seed-inl.h"
-#include "src/objects/hash-table-inl.h"
-#include "src/objects/heap-number-inl.h"
-#include "src/objects/js-array-buffer-inl.h"
 #include "src/objects/js-array-inl.h"
+#include "src/objects/js-promise-inl.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/ordered-hash-table-inl.h"
 #include "src/objects/promise-inl.h"
 #include "src/objects/smi.h"
-#include "src/objects/struct-inl.h"
 #include "src/objects/transitions-inl.h"
 #include "src/strings/char-predicates.h"
 #include "test/cctest/cctest-utils.h"
diff --git a/test/cctest/test-cpu-profiler.cc b/test/cctest/test-cpu-profiler.cc
index 76a891eaf04..ef0ad168e0e 100644
--- a/test/cctest/test-cpu-profiler.cc
+++ b/test/cctest/test-cpu-profiler.cc
@@ -40,20 +40,16 @@
 #include "src/base/strings.h"
 #include "src/codegen/compilation-cache.h"
 #include "src/codegen/source-position-table.h"
-#include "src/deoptimizer/deoptimizer.h"
+#include "src/deoptimizer/deoptimize-reason.h"
 #include "src/execution/embedder-state.h"
 #include "src/heap/spaces.h"
 #include "src/init/v8.h"
-#include "src/libplatform/default-platform.h"
-#include "src/libplatform/tracing/trace-event-listener.h"
 #include "src/libsampler/sampler.h"
 #include "src/logging/log.h"
 #include "src/objects/objects-inl.h"
-#include "src/profiler/cpu-profiler-inl.h"
+#include "src/profiler/cpu-profiler.h"
 #include "src/profiler/profiler-listener.h"
 #include "src/profiler/symbolizer.h"
-#include "src/profiler/tracing-cpu-profiler.h"
-#include "src/tracing/trace-event.h"
 #include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/heap/heap-utils.h"
diff --git a/test/cctest/test-debug-helper.cc b/test/cctest/test-debug-helper.cc
index 7d9b7a99718..3d0fb228d6c 100644
--- a/test/cctest/test-debug-helper.cc
+++ b/test/cctest/test-debug-helper.cc
@@ -8,7 +8,6 @@
 #include "src/execution/frames-inl.h"
 #include "src/flags/flags.h"
 #include "src/heap/read-only-spaces.h"
-#include "src/heap/spaces.h"
 #include "test/cctest/cctest.h"
 #include "tools/debug_helper/debug-helper.h"
 
diff --git a/test/cctest/test-debug.cc b/test/cctest/test-debug.cc
index 7c2c599d8c2..995f844740f 100644
--- a/test/cctest/test-debug.cc
+++ b/test/cctest/test-debug.cc
@@ -37,11 +37,8 @@
 #include "src/debug/debug-interface.h"
 #include "src/debug/debug.h"
 #include "src/deoptimizer/deoptimizer.h"
-#include "src/execution/frames.h"
 #include "src/execution/microtask-queue.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
-#include "src/snapshot/snapshot.h"
 #include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 
diff --git a/test/cctest/test-disasm-mips.cc b/test/cctest/test-disasm-mips.cc
index ae949e38137..5bb3ed46f83 100644
--- a/test/cctest/test-disasm-mips.cc
+++ b/test/cctest/test-disasm-mips.cc
@@ -28,12 +28,7 @@
 
 #include <stdlib.h>
 
-#include "src/init/v8.h"
-
-#include "src/codegen/macro-assembler.h"
-#include "src/debug/debug.h"
 #include "src/diagnostics/disasm.h"
-#include "src/diagnostics/disassembler.h"
 #include "src/execution/frames-inl.h"
 #include "test/cctest/cctest.h"
 
diff --git a/test/cctest/test-feedback-vector.cc b/test/cctest/test-feedback-vector.cc
index f65e0f1b006..68f95d83bcf 100644
--- a/test/cctest/test-feedback-vector.cc
+++ b/test/cctest/test-feedback-vector.cc
@@ -2,18 +2,14 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
+#include "test/cctest/test-feedback-vector.h"
 
 #include "src/api/api-inl.h"
-#include "src/codegen/macro-assembler.h"
-#include "src/debug/debug.h"
 #include "src/execution/execution.h"
-#include "src/handles/global-handles.h"
 #include "src/heap/factory.h"
 #include "src/objects/feedback-cell-inl.h"
 #include "src/objects/objects-inl.h"
-#include "test/cctest/test-feedback-vector.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-feedback-vector.h b/test/cctest/test-feedback-vector.h
index d50719f4af9..17c55e25cf8 100644
--- a/test/cctest/test-feedback-vector.h
+++ b/test/cctest/test-feedback-vector.h
@@ -5,6 +5,7 @@
 #ifndef V8_TEST_FEEDBACK_VECTOR_H_
 #define V8_TEST_FEEDBACK_VECTOR_H_
 
+#include "src/execution/isolate.h"
 #include "src/objects/feedback-vector.h"
 #include "src/objects/objects.h"
 #include "src/objects/shared-function-info.h"
diff --git a/test/cctest/test-field-type-tracking.cc b/test/cctest/test-field-type-tracking.cc
index d20eaa7ceb9..e5c8102c448 100644
--- a/test/cctest/test-field-type-tracking.cc
+++ b/test/cctest/test-field-type-tracking.cc
@@ -8,10 +8,7 @@
 
 #include "src/base/logging.h"
 #include "src/execution/execution.h"
-#include "src/handles/global-handles.h"
 #include "src/heap/factory-inl.h"
-#include "src/ic/stub-cache.h"
-#include "src/init/v8.h"
 #include "src/objects/field-type.h"
 #include "src/objects/heap-number-inl.h"
 #include "src/objects/internal-index.h"
@@ -22,7 +19,7 @@
 #include "src/objects/struct-inl.h"
 #include "src/objects/transitions.h"
 #include "src/utils/ostreams.h"
-#include "test/cctest/test-api.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-func-name-inference.cc b/test/cctest/test-func-name-inference.cc
index 410556093ad..dad3d414cde 100644
--- a/test/cctest/test-func-name-inference.cc
+++ b/test/cctest/test-func-name-inference.cc
@@ -27,8 +27,6 @@
 
 #include <memory>
 
-#include "src/init/v8.h"
-
 #include "src/api/api-inl.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
diff --git a/test/cctest/test-heap-profiler.cc b/test/cctest/test-heap-profiler.cc
index 192808fd017..25d1f79b4c9 100644
--- a/test/cctest/test-heap-profiler.cc
+++ b/test/cctest/test-heap-profiler.cc
@@ -42,7 +42,6 @@
 #include "src/debug/debug.h"
 #include "src/handles/global-handles.h"
 #include "src/heap/heap-inl.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/profiler/allocation-tracker.h"
 #include "src/profiler/heap-profiler.h"
diff --git a/test/cctest/test-helper-riscv64.cc b/test/cctest/test-helper-riscv64.cc
index e54d62ecec8..388e0ca0194 100644
--- a/test/cctest/test-helper-riscv64.cc
+++ b/test/cctest/test-helper-riscv64.cc
@@ -6,7 +6,6 @@
 
 #include "src/codegen/macro-assembler.h"
 #include "src/execution/isolate-inl.h"
-#include "src/init/v8.h"
 #include "test/cctest/cctest.h"
 
 namespace v8 {
diff --git a/test/cctest/test-helper-riscv64.h b/test/cctest/test-helper-riscv64.h
index 92e09bc5088..3cfc5427c69 100644
--- a/test/cctest/test-helper-riscv64.h
+++ b/test/cctest/test-helper-riscv64.h
@@ -7,11 +7,8 @@
 
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
-#include "src/diagnostics/disassembler.h"
 #include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
-#include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 
 #define PRINT_RES(res, expected_res, in_hex)                         \
diff --git a/test/cctest/test-ignition-statistics-extension.cc b/test/cctest/test-ignition-statistics-extension.cc
index 8c7d21ffee9..9a0b1336456 100644
--- a/test/cctest/test-ignition-statistics-extension.cc
+++ b/test/cctest/test-ignition-statistics-extension.cc
@@ -2,9 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "src/execution/isolate.h"
 #include "src/interpreter/bytecodes.h"
 #include "src/interpreter/interpreter.h"
-#include "test/cctest/test-api.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-inobject-slack-tracking.cc b/test/cctest/test-inobject-slack-tracking.cc
index 48718717c1d..1ecec21ac34 100644
--- a/test/cctest/test-inobject-slack-tracking.cc
+++ b/test/cctest/test-inobject-slack-tracking.cc
@@ -3,14 +3,13 @@
 // found in the LICENSE file.
 
 #include <stdlib.h>
+
 #include <sstream>
 #include <utility>
 
 #include "src/api/api-inl.h"
-#include "src/init/v8.h"
 #include "src/objects/heap-number-inl.h"
 #include "src/objects/objects-inl.h"
-
 #include "test/cctest/cctest.h"
 
 namespace v8 {
diff --git a/test/cctest/test-inspector.cc b/test/cctest/test-inspector.cc
index 420da2d68a4..e7f42074d46 100644
--- a/test/cctest/test-inspector.cc
+++ b/test/cctest/test-inspector.cc
@@ -7,7 +7,6 @@
 #include "include/v8-inspector.h"
 #include "include/v8-local-handle.h"
 #include "include/v8-primitive.h"
-#include "src/inspector/protocol/Runtime.h"
 #include "src/inspector/string-util.h"
 #include "src/inspector/v8-inspector-impl.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-javascript-arm64.cc b/test/cctest/test-javascript-arm64.cc
index df3984572db..792a5fb0629 100644
--- a/test/cctest/test-javascript-arm64.cc
+++ b/test/cctest/test-javascript-arm64.cc
@@ -27,16 +27,6 @@
 
 #include <limits.h>
 
-#include "src/init/v8.h"
-
-#include "src/api/api.h"
-#include "src/base/platform/platform.h"
-#include "src/codegen/compilation-cache.h"
-#include "src/execution/execution.h"
-#include "src/execution/isolate.h"
-#include "src/objects/objects-inl.h"
-#include "src/strings/unicode-inl.h"
-#include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 
 namespace v8 {
diff --git a/test/cctest/test-js-arm64-variables.cc b/test/cctest/test-js-arm64-variables.cc
index 46f2e20ba3c..24f484fdadf 100644
--- a/test/cctest/test-js-arm64-variables.cc
+++ b/test/cctest/test-js-arm64-variables.cc
@@ -29,16 +29,6 @@
 
 #include <limits.h>
 
-#include "src/init/v8.h"
-
-#include "src/api/api.h"
-#include "src/base/platform/platform.h"
-#include "src/codegen/compilation-cache.h"
-#include "src/execution/execution.h"
-#include "src/execution/isolate.h"
-#include "src/objects/objects-inl.h"
-#include "src/strings/unicode-inl.h"
-#include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 
 namespace v8 {
diff --git a/test/cctest/test-js-to-wasm.cc b/test/cctest/test-js-to-wasm.cc
index ff18adf51c7..469d9ad96d0 100644
--- a/test/cctest/test-js-to-wasm.cc
+++ b/test/cctest/test-js-to-wasm.cc
@@ -8,12 +8,10 @@
 #include "include/v8-local-handle.h"
 #include "include/v8-primitive.h"
 #include "include/v8-value.h"
-#include "src/api/api.h"
 #include "src/wasm/wasm-module-builder.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/compiler/node-observer-tester.h"
 #include "test/cctest/test-api.h"
-#include "test/common/wasm/flag-utils.h"
 #include "test/common/wasm/test-signatures.h"
 #include "test/common/wasm/wasm-macro-gen.h"
 
diff --git a/test/cctest/test-js-weak-refs.cc b/test/cctest/test-js-weak-refs.cc
index 39c3aee560e..1632d4e48ee 100644
--- a/test/cctest/test-js-weak-refs.cc
+++ b/test/cctest/test-js-weak-refs.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "src/execution/isolate.h"
-#include "src/execution/microtask-queue.h"
 #include "src/handles/handles-inl.h"
 #include "src/heap/factory-inl.h"
 #include "src/heap/heap-inl.h"
diff --git a/test/cctest/test-liveedit.cc b/test/cctest/test-liveedit.cc
index 38780351b76..679ec083b55 100644
--- a/test/cctest/test-liveedit.cc
+++ b/test/cctest/test-liveedit.cc
@@ -30,7 +30,6 @@
 #include "include/v8-function.h"
 #include "src/api/api-inl.h"
 #include "src/debug/liveedit.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "test/cctest/cctest.h"
 
diff --git a/test/cctest/test-local-handles.cc b/test/cctest/test-local-handles.cc
index f5fdc884453..ff38877a4da 100644
--- a/test/cctest/test-local-handles.cc
+++ b/test/cctest/test-local-handles.cc
@@ -5,8 +5,6 @@
 #include <memory>
 
 #include "src/api/api.h"
-#include "src/base/platform/condition-variable.h"
-#include "src/base/platform/mutex.h"
 #include "src/base/platform/semaphore.h"
 #include "src/common/assert-scope.h"
 #include "src/common/globals.h"
@@ -16,10 +14,8 @@
 #include "src/heap/heap.h"
 #include "src/heap/local-heap.h"
 #include "src/heap/parked-scope.h"
-#include "src/heap/safepoint.h"
 #include "src/objects/heap-number.h"
 #include "test/cctest/cctest.h"
-#include "test/cctest/heap/heap-utils.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-lockers.cc b/test/cctest/test-lockers.cc
index 385046f5059..287dc29e764 100644
--- a/test/cctest/test-lockers.cc
+++ b/test/cctest/test-lockers.cc
@@ -33,13 +33,8 @@
 #include "include/v8-function.h"
 #include "include/v8-locker.h"
 #include "src/base/platform/platform.h"
-#include "src/codegen/compilation-cache.h"
-#include "src/execution/execution.h"
-#include "src/execution/isolate.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/strings/unicode-inl.h"
-#include "src/utils/utils.h"
 #include "test/cctest/cctest.h"
 
 namespace {
diff --git a/test/cctest/test-log-stack-tracer.cc b/test/cctest/test-log-stack-tracer.cc
index ecdba08be88..445df286f5d 100644
--- a/test/cctest/test-log-stack-tracer.cc
+++ b/test/cctest/test-log-stack-tracer.cc
@@ -30,14 +30,11 @@
 #include <stdlib.h>
 
 #include "include/v8-function.h"
-#include "include/v8-profiler.h"
 #include "src/api/api-inl.h"
 #include "src/base/strings.h"
-#include "src/diagnostics/disassembler.h"
 #include "src/execution/frames.h"
 #include "src/execution/isolate.h"
 #include "src/execution/vm-state-inl.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/profiler/tick-sample.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-macro-assembler-arm.cc b/test/cctest/test-macro-assembler-arm.cc
index 26875793784..f66653a647c 100644
--- a/test/cctest/test-macro-assembler-arm.cc
+++ b/test/cctest/test-macro-assembler-arm.cc
@@ -31,7 +31,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-macro-assembler-arm64.cc b/test/cctest/test-macro-assembler-arm64.cc
index f43fad69965..7829fc8cebf 100644
--- a/test/cctest/test-macro-assembler-arm64.cc
+++ b/test/cctest/test-macro-assembler-arm64.cc
@@ -27,15 +27,11 @@
 
 #include <stdlib.h>
 
-#include "src/base/platform/platform.h"
 #include "src/codegen/arm64/assembler-arm64-inl.h"
 #include "src/codegen/macro-assembler-inl.h"
 #include "src/deoptimizer/deoptimizer.h"
-#include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
-#include "src/objects/smi.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
 #include "test/common/assembler-tester.h"
diff --git a/test/cctest/test-macro-assembler-loong64.cc b/test/cctest/test-macro-assembler-loong64.cc
index fb2295f496a..83b61966d02 100644
--- a/test/cctest/test-macro-assembler-loong64.cc
+++ b/test/cctest/test-macro-assembler-loong64.cc
@@ -34,7 +34,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-macro-assembler-mips.cc b/test/cctest/test-macro-assembler-mips.cc
index 11df7c410ba..feb86a1e45e 100644
--- a/test/cctest/test-macro-assembler-mips.cc
+++ b/test/cctest/test-macro-assembler-mips.cc
@@ -35,8 +35,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
-#include "src/init/v8.h"
-#include "src/objects/heap-number.h"
 #include "src/objects/js-array-inl.h"
 #include "src/objects/objects-inl.h"
 #include "src/utils/ostreams.h"
diff --git a/test/cctest/test-macro-assembler-mips64.cc b/test/cctest/test-macro-assembler-mips64.cc
index 56ecb3354da..dead68ee36f 100644
--- a/test/cctest/test-macro-assembler-mips64.cc
+++ b/test/cctest/test-macro-assembler-mips64.cc
@@ -34,7 +34,6 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-macro-assembler-riscv64.cc b/test/cctest/test-macro-assembler-riscv64.cc
index eab663b8f8b..7ce6e5d3dbb 100644
--- a/test/cctest/test-macro-assembler-riscv64.cc
+++ b/test/cctest/test-macro-assembler-riscv64.cc
@@ -34,10 +34,7 @@
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
-#include "src/init/v8.h"
-#include "src/objects/heap-number.h"
 #include "src/objects/objects-inl.h"
-#include "src/utils/ostreams.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/compiler/value-helper.h"
 #include "test/cctest/test-helper-riscv64.h"
diff --git a/test/cctest/test-macro-assembler-x64.cc b/test/cctest/test-macro-assembler-x64.cc
index c60bc10982f..94a66bc8d3f 100644
--- a/test/cctest/test-macro-assembler-x64.cc
+++ b/test/cctest/test-macro-assembler-x64.cc
@@ -27,13 +27,11 @@
 
 #include <stdlib.h>
 
-#include "src/base/platform/platform.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/codegen/x64/assembler-x64-inl.h"
 #include "src/deoptimizer/deoptimizer.h"
 #include "src/execution/simulator.h"
 #include "src/heap/factory.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
 #include "src/utils/ostreams.h"
diff --git a/test/cctest/test-orderedhashtable.cc b/test/cctest/test-orderedhashtable.cc
index f9117045838..72ff991e2cd 100644
--- a/test/cctest/test-orderedhashtable.cc
+++ b/test/cctest/test-orderedhashtable.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 #include <utility>
 
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/ordered-hash-table-inl.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-parsing.cc b/test/cctest/test-parsing.cc
index 513cbebf926..5b0d6c4daf5 100644
--- a/test/cctest/test-parsing.cc
+++ b/test/cctest/test-parsing.cc
@@ -31,25 +31,20 @@
 
 #include <memory>
 
-#include "include/v8-initialization.h"
-#include "include/v8-locker.h"
 #include "src/api/api-inl.h"
 #include "src/ast/ast-value-factory.h"
 #include "src/ast/ast.h"
 #include "src/base/enum-set.h"
 #include "src/base/strings.h"
-#include "src/codegen/compiler.h"
 #include "src/execution/execution.h"
 #include "src/execution/isolate.h"
 #include "src/flags/flags.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/objects.h"
 #include "src/parsing/parse-info.h"
 #include "src/parsing/parser.h"
 #include "src/parsing/parsing.h"
 #include "src/parsing/preparser.h"
-#include "src/parsing/rewriter.h"
 #include "src/parsing/scanner-character-streams.h"
 #include "src/parsing/token.h"
 #include "src/zone/zone-list-inl.h"  // crbug.com/v8/8816
diff --git a/test/cctest/test-profile-generator.cc b/test/cctest/test-profile-generator.cc
index 49e3942279f..702b7c2ff2a 100644
--- a/test/cctest/test-profile-generator.cc
+++ b/test/cctest/test-profile-generator.cc
@@ -32,7 +32,6 @@
 #include "src/api/api-inl.h"
 #include "src/base/strings.h"
 #include "src/init/v8.h"
-#include "src/logging/log.h"
 #include "src/objects/objects-inl.h"
 #include "src/profiler/cpu-profiler.h"
 #include "src/profiler/profile-generator-inl.h"
diff --git a/test/cctest/test-random-number-generator.cc b/test/cctest/test-random-number-generator.cc
index b1ef46465d7..c05c83db1fa 100644
--- a/test/cctest/test-random-number-generator.cc
+++ b/test/cctest/test-random-number-generator.cc
@@ -25,13 +25,11 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#include "src/base/utils/random-number-generator.h"
 #include "src/execution/isolate.h"
 #include "src/flags/flags.h"
-#include "src/init/v8.h"
 #include "test/cctest/cctest.h"
 
-#include "src/base/utils/random-number-generator.h"
-
 namespace v8 {
 namespace internal {
 
diff --git a/test/cctest/test-serialize.cc b/test/cctest/test-serialize.cc
index fc967e685af..811faff7f01 100644
--- a/test/cctest/test-serialize.cc
+++ b/test/cctest/test-serialize.cc
@@ -32,24 +32,17 @@
 #include "include/v8-function.h"
 #include "include/v8-locker.h"
 #include "src/api/api-inl.h"
-#include "src/codegen/assembler-inl.h"
 #include "src/codegen/compilation-cache.h"
 #include "src/codegen/compiler.h"
-#include "src/codegen/macro-assembler-inl.h"
 #include "src/codegen/script-details.h"
 #include "src/common/assert-scope.h"
-#include "src/debug/debug.h"
 #include "src/heap/heap-inl.h"
 #include "src/heap/parked-scope.h"
 #include "src/heap/read-only-heap.h"
 #include "src/heap/safepoint.h"
 #include "src/heap/spaces.h"
-#include "src/init/bootstrapper.h"
-#include "src/init/v8.h"
-#include "src/interpreter/interpreter.h"
 #include "src/numbers/hash-seed-inl.h"
 #include "src/objects/js-array-buffer-inl.h"
-#include "src/objects/js-array-inl.h"
 #include "src/objects/js-regexp-inl.h"
 #include "src/objects/objects-inl.h"
 #include "src/runtime/runtime.h"
diff --git a/test/cctest/test-shared-strings.cc b/test/cctest/test-shared-strings.cc
index 5682d7b5d3e..d37146cb4ae 100644
--- a/test/cctest/test-shared-strings.cc
+++ b/test/cctest/test-shared-strings.cc
@@ -9,7 +9,6 @@
 #include "src/heap/parked-scope.h"
 #include "src/objects/objects-inl.h"
 #include "test/cctest/cctest.h"
-#include "test/cctest/heap/heap-utils.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-smi-lexicographic-compare.cc b/test/cctest/test-smi-lexicographic-compare.cc
index 7b3e600c14c..929ed91ebdd 100644
--- a/test/cctest/test-smi-lexicographic-compare.cc
+++ b/test/cctest/test-smi-lexicographic-compare.cc
@@ -4,7 +4,6 @@
 
 #include <set>
 
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
 #include "test/cctest/cctest.h"
diff --git a/test/cctest/test-stack-unwinding-win64.cc b/test/cctest/test-stack-unwinding-win64.cc
index 4b30658ad31..a7b19af6c0c 100644
--- a/test/cctest/test-stack-unwinding-win64.cc
+++ b/test/cctest/test-stack-unwinding-win64.cc
@@ -7,8 +7,7 @@
 #include "include/v8-isolate.h"
 #include "include/v8-local-handle.h"
 #include "include/v8-template.h"
-#include "src/base/win32-headers.h"
-#include "src/init/v8.h"
+#include "src/base/macros.h"
 #include "test/cctest/cctest.h"
 
 #if defined(V8_OS_WIN_X64)
diff --git a/test/cctest/test-strings.cc b/test/cctest/test-strings.cc
index b29a8a06071..9abce9f032d 100644
--- a/test/cctest/test-strings.cc
+++ b/test/cctest/test-strings.cc
@@ -32,7 +32,6 @@
 
 #include <stdlib.h>
 
-#include "include/v8-initialization.h"
 #include "include/v8-json.h"
 #include "src/api/api-inl.h"
 #include "src/base/platform/elapsed-timer.h"
@@ -40,9 +39,7 @@
 #include "src/execution/messages.h"
 #include "src/heap/factory.h"
 #include "src/heap/heap-inl.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
-#include "src/strings/unicode-decoder.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/heap/heap-utils.h"
 
diff --git a/test/cctest/test-swiss-name-dictionary-infra.h b/test/cctest/test-swiss-name-dictionary-infra.h
index 60ac78477a4..ed0c03a0f1a 100644
--- a/test/cctest/test-swiss-name-dictionary-infra.h
+++ b/test/cctest/test-swiss-name-dictionary-infra.h
@@ -8,13 +8,9 @@
 #include <memory>
 #include <utility>
 
-#include "src/codegen/code-stub-assembler.h"
-#include "src/init/v8.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/swiss-name-dictionary-inl.h"
 #include "test/cctest/cctest.h"
-#include "test/cctest/compiler/code-assembler-tester.h"
-#include "test/cctest/compiler/function-tester.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-sync-primitives-arm.cc b/test/cctest/test-sync-primitives-arm.cc
index 2660c693f51..a326e22714b 100644
--- a/test/cctest/test-sync-primitives-arm.cc
+++ b/test/cctest/test-sync-primitives-arm.cc
@@ -25,15 +25,11 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#include "src/init/v8.h"
-#include "test/cctest/assembler-helper-arm.h"
-#include "test/cctest/cctest.h"
-
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
-#include "src/diagnostics/disassembler.h"
 #include "src/execution/simulator.h"
-#include "src/heap/factory.h"
+#include "test/cctest/assembler-helper-arm.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-sync-primitives-arm64.cc b/test/cctest/test-sync-primitives-arm64.cc
index c2b46fc8b3d..6efec4948eb 100644
--- a/test/cctest/test-sync-primitives-arm64.cc
+++ b/test/cctest/test-sync-primitives-arm64.cc
@@ -25,13 +25,10 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
-
 #include "src/codegen/macro-assembler-inl.h"
 #include "src/execution/arm64/simulator-arm64.h"
-#include "src/heap/factory.h"
 #include "src/objects/objects-inl.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-temporal-parser.cc b/test/cctest/test-temporal-parser.cc
index 82d1a4626d9..710348cb322 100644
--- a/test/cctest/test-temporal-parser.cc
+++ b/test/cctest/test-temporal-parser.cc
@@ -3,9 +3,7 @@
 // found in the LICENSE file.
 
 #include "src/execution/isolate.h"
-#include "src/handles/global-handles.h"
 #include "src/heap/factory-inl.h"
-#include "src/init/v8.h"
 #include "src/temporal/temporal-parser.h"
 #include "test/cctest/cctest.h"
 
diff --git a/test/cctest/test-transitions.cc b/test/cctest/test-transitions.cc
index c575b5d8e91..0cee7dc6f92 100644
--- a/test/cctest/test-transitions.cc
+++ b/test/cctest/test-transitions.cc
@@ -2,20 +2,19 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "test/cctest/test-transitions.h"
+
 #include <stdlib.h>
-#include <utility>
 
-#include "src/init/v8.h"
+#include <utility>
 
 #include "src/codegen/compilation-cache.h"
 #include "src/execution/execution.h"
-#include "src/handles/global-handles.h"
 #include "src/heap/factory.h"
 #include "src/objects/field-type.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/transitions-inl.h"
 #include "test/cctest/cctest.h"
-#include "test/cctest/test-transitions.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-typedarrays.cc b/test/cctest/test-typedarrays.cc
index 0134befedd2..72484b04559 100644
--- a/test/cctest/test-typedarrays.cc
+++ b/test/cctest/test-typedarrays.cc
@@ -4,13 +4,8 @@
 
 #include <stdlib.h>
 
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
-
 #include "src/execution/protectors-inl.h"
-#include "src/heap/heap.h"
-#include "src/objects/objects-inl.h"
-#include "src/objects/objects.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-unscopables-hidden-prototype.cc b/test/cctest/test-unscopables-hidden-prototype.cc
index e4e2a08e1f6..95666e3acf2 100644
--- a/test/cctest/test-unscopables-hidden-prototype.cc
+++ b/test/cctest/test-unscopables-hidden-prototype.cc
@@ -5,7 +5,6 @@
 #include <stdlib.h>
 
 #include "include/v8-function.h"
-#include "src/init/v8.h"
 #include "test/cctest/cctest.h"
 
 namespace {
diff --git a/test/cctest/test-usecounters.cc b/test/cctest/test-usecounters.cc
index 817f5fb3518..69227ee0ac3 100644
--- a/test/cctest/test-usecounters.cc
+++ b/test/cctest/test-usecounters.cc
@@ -2,8 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "src/init/v8.h"
-
 #include "test/cctest/cctest.h"
 
 namespace v8 {
diff --git a/test/cctest/test-utils-arm64.cc b/test/cctest/test-utils-arm64.cc
index ae02d547e58..fd09ef3c2f1 100644
--- a/test/cctest/test-utils-arm64.cc
+++ b/test/cctest/test-utils-arm64.cc
@@ -29,10 +29,7 @@
 
 #include "src/base/template-utils.h"
 #include "src/codegen/arm64/assembler-arm64-inl.h"
-#include "src/codegen/arm64/utils-arm64.h"
 #include "src/codegen/macro-assembler-inl.h"
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-utils-arm64.h b/test/cctest/test-utils-arm64.h
index 21b2123eead..3f0dd6f9d8f 100644
--- a/test/cctest/test-utils-arm64.h
+++ b/test/cctest/test-utils-arm64.h
@@ -28,10 +28,7 @@
 #ifndef V8_ARM64_TEST_UTILS_ARM64_H_
 #define V8_ARM64_TEST_UTILS_ARM64_H_
 
-#include "src/codegen/arm64/utils-arm64.h"
 #include "src/codegen/macro-assembler.h"
-#include "src/init/v8.h"
-#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/test-utils.cc b/test/cctest/test-utils.cc
index b06669a9944..479fde1ed00 100644
--- a/test/cctest/test-utils.cc
+++ b/test/cctest/test-utils.cc
@@ -29,11 +29,8 @@
 
 #include <vector>
 
-#include "include/v8-initialization.h"
 #include "src/api/api-inl.h"
 #include "src/base/bit-field.h"
-#include "src/base/platform/platform.h"
-#include "src/init/v8.h"
 #include "src/numbers/conversions.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/collector.h"
diff --git a/test/cctest/trace-extension.h b/test/cctest/trace-extension.h
index 0ddc0f0e8c1..a7aaa5e4b54 100644
--- a/test/cctest/trace-extension.h
+++ b/test/cctest/trace-extension.h
@@ -29,7 +29,6 @@
 #define V8_TEST_CCTEST_TRACE_EXTENSION_H_
 
 #include "include/v8-extension.h"
-#include "src/common/globals.h"
 
 namespace v8 {
 
diff --git a/test/common/wasm/wasm-macro-gen.h b/test/common/wasm/wasm-macro-gen.h
index b5be4dabaf1..feb42685404 100644
--- a/test/common/wasm/wasm-macro-gen.h
+++ b/test/common/wasm/wasm-macro-gen.h
@@ -7,8 +7,6 @@
 
 #include "src/wasm/wasm-opcodes.h"
 
-#include "src/zone/zone-containers.h"
-
 #define U32_LE(v)                                    \
   static_cast<byte>(v), static_cast<byte>((v) >> 8), \
       static_cast<byte>((v) >> 16), static_cast<byte>((v) >> 24)
-- 
2.35.1

